spec/graphql/mutations/create_user_spec.rb

```diff
@@ -2,69 +2,56 @@
+    let(:parsed_response) { JSON.parse(response.body, symbolize_names: true) }
 
-        expect(response).to be_successful
-        parsed_response = JSON.parse(response.body, symbolize_names: true)
+    context 'happy path' do
+      before { post '/graphql', params: { query: mutation('Floofything') } }
 
+      it 'can create a user from a mutation' do
+        expect(response).to be_successful
         expect(parsed_response[:data][:createUser][:user][:username]).to eq('Floofything')
         expect(parsed_response[:data][:createUser][:errors]).to eq([])
       end
     end
 
-    describe 'sad paths' do
-      it 'returns an appropriate error if user field is blank' do
-        post '/graphql', params: { query: bad_mutation }
+    context 'sad paths' do
+      context 'when user field is blank' do
+        before { post '/graphql', params: { query: mutation('') } }
 
-        expect(response).to be_successful
-        parsed_response = JSON.parse(response.body, symbolize_names: true)
-
-        expect(parsed_response[:data][:createUser]).to eq(nil)
-        expect(parsed_response[:errors].first[:message]).to eq('Cannot return null for non-nullable field CreateUserPayload.user')
+        it 'returns an appropriate error' do
+          expect(response).to be_successful
+          expect(parsed_response[:data][:createUser]).to be_nil
+          expect(parsed_response[:errors].first[:message]).to eq('Cannot return null for non-nullable field CreateUserPayload.user')
+        end
       end
 
-      it 'returns an appropriate error if username is not unique' do
-        create(:user, username: 'Floofything')
-        post '/graphql', params: { query: mutation }
-
-        expect(response).to be_successful
-        parsed_response = JSON.parse(response.body, symbolize_names: true)
-
-        expect(parsed_response[:data][:createUser]).to eq(nil)
-        expect(parsed_response[:errors].first[:message]).to eq('Cannot return null for non-nullable field CreateUserPayload.user')
+      context 'when username is not unique' do
+        before do
+          create(:user, username: 'Floofything')
+          post '/graphql', params: { query: mutation('Floofything') }
+        end
+
+        it 'returns an appropriate error' do
+          expect(response).to be_successful
+          expect(parsed_response[:data][:createUser]).to be_nil
+          expect(parsed_response[:errors].first[:message]).to eq('Cannot return null for non-nullable field CreateUserPayload.user')
+        end
       end
     end
   end
-end
-
-def mutation
-  <<~GQL
-    mutation {
-    createUser(input:{
-      username: "Floofything"
-      }) {
-      user {
-            username
-          }
-        errors
-        }
-    }
-  GQL
-end
 
-def bad_mutation
-  <<~GQL
-    mutation {
-    createUser(input:{
-      username: ""
-      }) {
-      user {
+  def mutation(username)
+    <<~GQL
+      mutation {
+        createUser(input:{
+          username: "#{username}"
+        }) {
```

Comment:
where does `#{username}` come from?

Reply:
It's passed through as an argument

```ruby
before { post '/graphql', params: { query: mutation('Floofything') } }
``` 

Or in the case of a blank query, nothing is passed through

```ruby
before { post '/graphql', params: { query: mutation('') } }
```

Reply:
In fact, most of the DRY'd up code in this PR was done by implementing arguments into the test queries.